builder/struct.go

```diff
@@ -37,110 +39,246 @@ func (s *Struct) Assign(gen Generator, ctx *MethodContext, assignTo *AssignTo, s
+			if err != nil {
+				return nil, false, err
+			}
+			usedSourceID = usedSourceID || fieldUsedSourceID
+			stmt = append(stmt, fieldStmt...)
 		}
-		if !targetField.Exported() && ctx.Conf.IgnoreUnexported {
-			continue
+	}
+	return stmt, usedSourceID, nil
+}
+
+func fieldGen(gen Generator, ctx *MethodContext, assignTo *AssignTo, sourceID *xtype.JenID, source, target *xtype.Type, errPath ErrorPath, additionalFieldSources []xtype.FieldSources, definedFields map[string]struct{}, missing map[string]*Error, targetField *types.Var) ([]jen.Code, bool, *Error) {
+	delete(definedFields, targetField.Name())
+
+	fieldMapping := ctx.Field(target, targetField.Name())
+
+	if fieldMapping.Ignore {
+		return nil, false, nil
+	}
+	if !targetField.Exported() && ctx.Conf.IgnoreUnexported {
+		return nil, false, nil
+	}
+
+	if !xtype.Accessible(targetField, ctx.OutputPackagePath) {
+		cause := unexportedStructError(targetField.Name(), source.String, target.String)
+		return nil, false, NewError(cause).Lift(&Path{
+			Prefix:     ".",
+			SourceID:   "???",
+			TargetID:   targetField.Name(),
+			TargetType: targetField.Type().String(),
+		})
+	}
+
+	implicitName := targetField.Name()
+	tempStmt := []jen.Code{}
+	targetFieldType := xtype.TypeOf(targetField.Type())
+	targetFieldPath := errPath.Field(targetField.Name())
+	assignVar := assignTo.Stmt.Clone().Dot(targetField.Name())
+	setterStmt := []jen.Code{}
+
+	setterSignature, ok := targetField.Type().(*types.Signature)
+	if ok && setterSignature.Recv() != nil {
+		matches := ctx.Conf.SettersRegex.FindStringSubmatch(targetField.Name())
+		if len(matches) == 2 && matches[0] == targetField.Name() {
+			implicitName = matches[1]
+		} else if fieldMapping.Source == "" && fieldMapping.Function == nil {
+			return nil, false, nil
 		}
 
-		if !xtype.Accessible(targetField, ctx.OutputPackagePath) {
-			cause := unexportedStructError(targetField.Name(), source.String, target.String)
-			return nil, NewError(cause).Lift(&Path{
+		if setterSignature.Params().Len() != 1 {
+			return nil, false, NewError(fmt.Sprintf("Setter method %s must have exactly one parameter.", targetField.Name())).Lift(&Path{
 				Prefix:     ".",
 				SourceID:   "???",
 				TargetID:   targetField.Name(),
-				TargetType: targetField.Type().String(),
+				TargetType: "???",
 			})
 		}
 
-		targetFieldType := xtype.TypeOf(targetField.Type())
-		targetFieldPath := errPath.Field(targetField.Name())
+		setParam := setterSignature.Params().At(0)
+		targetField := types.NewVar(token.NoPos, nil, targetField.Name(), setParam.Type())
+		targetFieldType = xtype.TypeOf(targetField.Type())
+		var err *Error
+		tempStmt, assignVar, err = buildTargetVar(gen, ctx, sourceID, source, targetFieldType, targetFieldPath)
+		if err != nil {
+			return nil, false, err
+		}
+		setterStmt = []jen.Code{assignTo.Stmt.Clone().Dot(targetField.Name()).Call(assignVar.Clone())}
```

Comment:
This should use the Generator#CallMethod function or a new method that does call a method without target return. This will allow to return error values on the setters, and also gives context support.